metrics.go

```diff
@@ -21,83 +21,82 @@ import (
+	SamplerUpdateFailure metrics.Counter `metric:"sampler_updates" tags:"result=err" help:"Number of times the Sampler failed to update sampling strategy"`
 
 	// Number of times baggage was successfully written or updated on spans.
-	BaggageUpdateSuccess metrics.Counter `metric:"baggage_updates" tags:"result=ok"`
+	BaggageUpdateSuccess metrics.Counter `metric:"baggage_updates" tags:"result=ok" help:"Number of times baggage was successfully written or updated on spans"`
 
 	// Number of times baggage failed to write or update on spans.
-	BaggageUpdateFailure metrics.Counter `metric:"baggage_updates" tags:"result=err"`
+	BaggageUpdateFailure metrics.Counter `metric:"baggage_updates" tags:"result=err" help:"Number of times baggage failed to write or update on spans"`
 
 	// Number of times baggage was truncated as per baggage restrictions.
-	BaggageTruncate metrics.Counter `metric:"baggage_truncations"`
+	BaggageTruncate metrics.Counter `metric:"baggage_truncations" help:"Number of times baggage was truncated as per baggage restrictions"`
 
 	// Number of times baggage restrictions were successfully updated.
-	BaggageRestrictionsUpdateSuccess metrics.Counter `metric:"baggage_restrictions_updates" tags:"result=ok"`
+	BaggageRestrictionsUpdateSuccess metrics.Counter `metric:"baggage_restrictions_updates" tags:"result=ok" help:"Number of times baggage restrictions were successfully updated"`
 
 	// Number of times baggage restrictions failed to update.
-	BaggageRestrictionsUpdateFailure metrics.Counter `metric:"baggage_restrictions_updates" tags:"result=err"`
+	BaggageRestrictionsUpdateFailure metrics.Counter `metric:"baggage_restrictions_updates" tags:"result=err" help:"Number of times baggage restrictions failed to update"`
 
 	// Number of times debug spans were throttled.
-	ThrottledDebugSpans metrics.Counter `metric:"throttled_debug_spans"`
+	ThrottledDebugSpans metrics.Counter `metric:"throttled_debug_spans" help:"Number of times debug spans were throttled"`
 
 	// Number of times throttler successfully updated.
-	ThrottlerUpdateSuccess metrics.Counter `metric:"throttler_updates" tags:"result=ok"`
+	ThrottlerUpdateSuccess metrics.Counter `metric:"throttler_updates" tags:"result=ok" help:"Number of times throttler successfully updated"`
 
 	// Number of times throttler failed to update.
-	ThrottlerUpdateFailure metrics.Counter `metric:"throttler_updates" tags:"result=err"`
+	ThrottlerUpdateFailure metrics.Counter `metric:"throttler_updates" tags:"result=err" help:"Number of times throttler failed to update"`
 }
 
 // NewMetrics creates a new Metrics struct and initializes it.
 func NewMetrics(factory metrics.Factory, globalTags map[string]string) *Metrics {
 	m := &Metrics{}
-	// TODO the namespace "jaeger" should be configurable (e.g. in all-in-one "jaeger-client" would make more sense)
-	metrics.MustInit(m, factory.Namespace(metrics.NSOptions{Name: "jaeger"}), globalTags)
```

Comment:
I think removing the `jaeger` prefix is a major breaking change, that could even lead to applications not starting since it's possible that the same metric names will be defined elsewhere.

Reply:
That part of the metric name was hardcoded, and then there was a prefix - so for example `jaeger_client_jaeger_`. Therefore I think it is unlikely that removing this hardcoded second `jaeger` in the name would conflict with existing metric names in any system that uses Jaeger.

Reply:
where is `jaeger_client` prefix provided?

Reply:
Those are places outside of this lib; that was exactly my point - if the end user was already providing a prefix, they're likely to be ok, but many may not have provided a prefix since there was already a built-in prefix in the client (which TODO was suggesting to make configurable).

Reply:
What about changing jaeger to tracer within the client lib?

Reply:
Ok will look at making it configurable, and perhaps default it to `jaeger_tracer_` to make sure it does not conflict with existing stored metrics. Then within the backend, by default we don't change the prefix. Sound ok?

Reply:
As a short term non-breakingAPI change, how about changing the current (go client) default prefix from [`jaeger` to `jaeger_tracer`](https://github.com/jaegertracing/jaeger-client-go/blob/master/metrics.go#L100), to be consistent with the updated [java client](https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/src/main/java/io/jaegertracing/internal/metrics/Metrics.java#L27)? Currently the java client provides a constructor for changing the prefix, but isn't used at the moment (i.e. it is not exposed in such a way that the application could change it).

So, if an app doesn't create its own namespace, all of the go client metrics would be `jaeger_tracer_....`. We then just need to change the backend so that it doesn't supply a sub-namespace (currently `jaeger_client`) and instead just provides the top level namespace.

As a separate issue, for all clients, we could then decide how a metrics prefix should be configurable by the end user.

Reply:
> from jaeger to jaeger_tracer, 

+1

Reply:
Done